### PR TITLE
[TE] Authorize service account to prevent config modification

### PR DESCRIPTION
**Pull request by akshayrai**

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/YamlResource.java [hunk 1]
```diff
@@ -375,6 +398,43 @@ void updateDetectionPipeline(long detectionID, String payload, long startTime, l
     }
   }
 
+  private boolean isServiceAccount(ThirdEyePrincipal user) {
+    List<Predicate> predicates = new ArrayList<>();
+    predicates.add(Predicate.EQ("sessionKey", user.getSessionKey()));
+    predicates.add(Predicate.EQ("principalType", "SERVICE"));
```
Comment (jihaozh) [inline thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/YamlResource.java]: define these strings as private static finals?

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/YamlResource.java [hunk 2]
```diff
@@ -222,12 +230,23 @@ private Response processServerErrorResponse(String type, String operation, Strin
...
+    responseMessage.put(type + "Msg", "Authorization Error!");
+    responseMessage.put(type + "Msg-moreInfo", "Configure owners property in " + type + " config");
+    return Response.status(Response.Status.UNAUTHORIZED).entity(responseMessage).build();
+  }
+
   @POST
   @Path("/create-alert")
   @Produces(MediaType.APPLICATION_JSON)
   @Consumes(MediaType.APPLICATION_JSON)
+  @PermitAll
```
Comment (jihaozh) [inline thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/YamlResource.java]: Why `PermitAll`?

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/YamlResource.java [hunk 3]
```diff
@@ -375,6 +398,43 @@ void updateDetectionPipeline(long detectionID, String payload, long startTime, l
     }
   }
 
+  private boolean isServiceAccount(ThirdEyePrincipal user) {
+    List<Predicate> predicates = new ArrayList<>();
+    predicates.add(Predicate.EQ("sessionKey", user.getSessionKey()));
+    predicates.add(Predicate.EQ("principalType", "SERVICE"));
+
+    List<SessionDTO> sessionDTO = this.sessionDAO.findByPredicate(Predicate.AND(predicates.toArray(new Predicate[0])));
+    return sessionDTO != null && !sessionDTO.isEmpty();
```
Comment (jihaozh) [inline thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/YamlResource.java]: Do we check the session expiration time?

Comment (akshayrai) [reply thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/YamlResource.java]: Yes, session expiry is checked separately by the auth module.

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/YamlResource.java (hunk 2 above)
Comment (akshayrai) [reply thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/YamlResource.java]: This doesn't really matter. It comes into play when we define security roles.

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/YamlResource.java (hunk 1 above)
Comment (akshayrai) [reply thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/YamlResource.java]: done

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/YamlResource.java [hunk 4]
```diff
@@ -375,6 +398,43 @@ void updateDetectionPipeline(long detectionID, String payload, long startTime, l
...
+  }
+
+  private void validateConfigOwner(ThirdEyePrincipal user, List<String> owners) {
+    if (owners == null || !owners.contains(user.getName())) {
```
Comment (xiaohui-sun) [inline thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/YamlResource.java]: We need to update existing alerts created by service accounts.

Comment (akshayrai) [reply thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/YamlResource.java]: Good point. I could find 6 alerts, all created by career-comms service account. I will update them during deployment.

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/YamlResource.java (hunk 3 above)
Comment (jihaozh) [reply thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/YamlResource.java]: make sense

**Pull request merged**